### DISCUSSION TITLE
Multiple enviroments

### DISCUSSION THREAD

**Discussion by nachitox** [SublimeLinter/SublimeLinter#1890] [Q&A]
I have a project with multiple repos.
Each repo has its own environment/packages.

My settings point to the main repo:

```
{
	"linters": {
		"pylint": {
			"filter_errors": [
				"C0114", // missing-module-docstring
				"E1101" // no-member
			],
			"python": "/.../main-repo/venv/bin/python3"
		}
	}
}
```

but when browsing another repo, I get
> pylint: error E0401 - Unable to import 'flask' (import-error)

Is there a way of adding multiple python paths? Something like this (it fails):

```
{
	"linters": {
		"pylint": {
			"filter_errors": [
				"C0114", // missing-module-docstring
				"E1101" // no-member
			],
			"python": [
				"/.../main-repo/venv/bin/python3",
				"/.../other-repo/venv/bin/python3"
			]
		}
	}
}
```

Thanks

**Comment by kaste** [comment]
This works with Sublime Text projects.  In each project file you store a separate config for SublimeLinter.  Under the `settings` key.  E.g. a typical project file, disabling SublimeLinter for the folder the project lives in:

```

{
    "folders": [
        {
            "path": "."
        },
    ],

    "settings": {
        "SublimeLinter.enabled?": false,
    }
}

```

**Reply by nachitox** [reply]
Do you suggest I should have a project for each repo?
Some of them are related (one call another) so having them on the same project make things easier.

**Reply by kaste** [reply]
Yeah, that's totally not ideal.  Two things don't work here.  First, you use `venv` but we only look for `.venv`.  Of course this is arbitrary and we probably should *also* look for your spelling.  Second, we don't walk up the directories starting from the current file, searching for the nearest possible *root*.  (We do this for node based linters, for example *eslint*.)  We only search in the working dir root, which is typically the top folder you see in Sublime's sidebar.  But you can open multiple "top" folders in the side bar and we will probably use the correct one.  

There is also an open issue around this.  Let me look this up.

**Reply by kaste** [reply]
It's https://github.com/SublimeLinter/SublimeLinter/issues/1883

**Comment by kaste** [comment]
The latest SublimeLinter version now traverses up starting with the filename and looks for virtual environments in `venv` dirs too.  So maybe it just works and you can remove the `python` setting.
